src/passkey_child/passkey_child_common.c

```diff
@@ -319,3 +369,99 @@ register_key(struct passkey_data *data)
 
     return ret;
 }
+
+errno_t
+authenticate(struct passkey_data *data)
+{
+    fido_assert_t *assert = NULL;
+    fido_dev_info_t *dev_list = NULL;
+    fido_dev_t *dev = NULL;
+    size_t dev_number = 0;
+    errno_t ret;
+
+    dev_list = fido_dev_info_new(DEVLIST_SIZE);
+    if (dev_list == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "fido_dev_info_new failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    ret = list_devices(dev_list, &dev_number);
+    if (ret != EOK) {
+        goto done;
+    }
+
+    if (dev_number == 0) {
+        DEBUG(SSSDBG_OP_FAILURE, "No device found. Aborting.\n");
+        ret = ENOENT;
+        goto done;
+    }
+
+    assert = fido_assert_new();
+    if (assert == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "fido_assert_new failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    ret = prepare_assert(data, assert);
+    if (ret != FIDO_OK) {
+        goto done;
+    }
+
+    for (int i = 0; i < dev_number; i++) {
+        DEBUG(SSSDBG_TRACE_FUNC,
+              "Attempting authentication with device number %d.\n", i + 1);
+
+        ret = decode_public_key(data);
+        if (ret != FIDO_OK) {
+            goto done;
+        }
+
+        ret = get_device_assert(data, dev_list, dev_number, assert,
+                                &dev);
+        if (ret != FIDO_OK) {
+            goto done;
+        }
+
+        ret = get_device_options(data, dev);
+        if (ret != FIDO_OK) {
+            goto done;
+        }
+
+        ret = set_assert_options(data, assert);
+        if (ret != FIDO_OK) {
+            DEBUG(SSSDBG_OP_FAILURE, "Failed to reset assert options.\n");
+            goto done;
+        }
+
+        ret = set_assert_client_data_hash(assert);
+        if (ret != FIDO_OK) {
+            DEBUG(SSSDBG_OP_FAILURE, "Failed to reset client data hash.\n");
+            goto done;
+        }
+
+        ret = verify_assert(data, dev, assert);
+        if (ret == FIDO_OK) {
+            // End the loop only if the authentication has been successful
```

Comment:
sssd coding style states Do not use C99 line comments (// line comment)

Reply:
Fixed